### ISSUE TITLE
Lockfile not always getting deleted

### ISSUE THREAD

**Issue by SpeedHighway** [PortOfPortland/terraform-provider-windns#18]
I was having an issue that seemed similar to existing issues where the windns provider was perpetually sitting on "Still creating...."

For me this happens on a build server, so I checked the next day.  After 16 hours, it was still sitting on "Still creating..."
Looking at the code, the only real loop in the application is checking for the lock file to be deleted.  I used process monitor to find out where it was looking for the file and after opening the temp folder for my build agent's user, found 567 lock files all named "terraform-windns" with a random string of numbers after it.  This seems to match up to the code.  

The problem, however, is that none of these files had ever gotten deleted and we were apparently hitting the random chance of those strings occurring again.  As such, this provider just sits in a loop, forever, waiting for files to delete that never will.  Upon deleting the lock files, myself, the provider went on and completed as if nothing was wrong.  

After investigating further, I realized that this line gets hit when it starts:
https://github.com/PortOfPortland/terraform-provider-windns/blob/901bf19d09e1d96737131308a2b0f90c99cfc1e1/windns/provider.go#L76

As you can see here:
https://golang.org/src/io/ioutil/tempfile.go?s=1429:1487#L41
this creates files.

However, if there is nothing to be done (for example, windns is already created via the terraform state) then the lockfile is just left, forever, and never deleted.

There is an os.Remove command after that, but it doesn't seem to be accomplishing anything as the files are staying.
While I've been typing this, I've obtained 10 more windns lock files on this server, without any dns entries being created, because no work was needed.

Edit: After doing a test on re-running a release that's already created, it is creating exactly 5 lock files, that never get deleted, for each run.  (I've watched process manager during an actual run where it hits the create and the lock files created there are deleted.  But when doing only the config method, there's never a call to delete in process manager.)

**Comment by rayterrill** [comment]
@SpeedHighway Really appreciate this detailed analysis mate. I believe this code was a contribution we accepted - wrapped up some other efforts last week and I'll dig into this and see if we can get this fixed this week.

**Comment by rayterrill** [comment]
@SpeedHighway What OS are you running this on? I'm running this on Linux (Ubuntu 16.04) and I cannot replicate. Assuming Windows but which OS so I can replicate.

**Comment by rayterrill** [comment]
Nevermind - I was able to replicate this on Windows. I added some additional debugging to a DEV build, getting a typical Windows error trying to delete the file:
```
terraform-windns866116175: The process cannot access the file because it is being used by another process.
```

**Comment by rayterrill** [comment]
@SpeedHighway Can you test this with the new v0.5.2 build I just uploaded to Github? This fixes a few bugs with the way that tempfile locking was implemented. https://github.com/PortOfPortland/terraform-provider-windns/releases/tag/v0.5.2

**Comment by SpeedHighway** [comment]
I will try to test this out tomorrow if I am able.  Thanks for looking at it. :)

**Comment by SpeedHighway** [comment]
I did a verification of the above scenario in which no changes to WinDNS needed to be done and it worked perfectly with no leftover lock files, thanks!

(Sorry, I typed this out the other day with the intent of testing one that DID require WinDNS changes, but haven't had the chance, yet, so I'll just submit the comment that I have tested)

**Comment by rayterrill** [comment]
Sounds good. I'll go ahead and get this built out into a full release. Thank you for confirming, @SpeedHighway!

**Comment by rayterrill** [comment]
@SpeedHighway released as v0.5.3. Thank you for the detailed report!

**Comment by SpeedHighway** [comment]
I was also able to verify that it worked correctly when adding new DNS entries.  Kudos!
